Task 3/Task3.2/DynamicArray.cs

```diff
@@ -0,0 +1,184 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.Serialization;
+using System.Text;
+
+namespace Task3._2
+{
+    class DynamicArray<T>:ICollection<T>,IEnumerable<T>
+    {
+        public int index = -1;
+        private int capacity = 8;
+        private int length = 0;
+        private T[] data;
+        public DynamicArray()
+        {
+            Array.Resize(ref data, Capacity);
+        }
+        public T this[int index]
+        {
+            get { return data[index]; }
+            set { data[index] = value; }
+        }
+        public int Count
+        {
+            get
+            {
+                return data.Length;
+            }
+        }
+        public int Capacity
+        {
+            get
+            {
+                return capacity;
+            }
+            set
+            {
+                if (capacity >= 0)
+                    capacity = value;
+            }
+        }
+        public int Length
+        {
+            get
+            {
+                return length;
+            }
+            set
+            {
+                length = value;
+            }
+        }
```

Comment:
```suggestion
        public int Length
        {
            get
            {
                return data.Length;
            }
        }
        public int Capacity
        {
            get
            {
                return capacity;
            }
            set
            {
                if (capacity >= Length)
                    capacity = value;
            }
        }
```
1) Не совсем понял зачем тебе тут три свойства для описания размеров коллекции.
2) Не корректно давать пользователю давать возможность менять Length т.к. это свойство все таки характеризует кол-во элементов в коллекции и вилять он должен на это свойство именно добавление/удалением элементов.
3) Нельзя давать возможность задать емкость коллекции меньше, чем кол-во элементов в нем, чтобы избежать потери данных.

Reply:
1. Да я просто хотел, чтобы как у всех нормальных коллекций было свойство Count.🤔
2. Согласен. Поторопился.
3. Я просто неправильно сначала понял дополнительный пункт про ручное изменение ёмкости.

Reply:
2. Может быть, я тогда недостаточно хорошо подумал, но теперь AddRange сломался.

Reply:
2. Я очень извиняюсь. Я из-за спешки умудрился сломать геттер, а там и AddRange. Теперь всё работает.👌